src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/DynamicInvokeInfo.cs

```diff
@@ -838,6 +838,11 @@ private unsafe object ReturnTransform(ref byte byref, bool wrapInTargetInvocatio
                 Debug.Assert(type.IsPointer);
                 obj = Pointer.Box((void*)Unsafe.As<byte, IntPtr>(ref byref), type);
             }
+            else if ((_returnTransform & Transform.FunctionPointer) != 0)
```

Comment:
Without this change, a checked build would throw in the `else` statement below where it tried to box `_returnType`.